go/packages/dev-mode/main_test.go

```diff
@@ -0,0 +1,287 @@
+	payload := fmt.Sprintf(`{
+    "type": "platform.runtimeDone",
+		"record": {
+			"requestId": "%s",
+			"status": "success"
+		}
+	}`, requestId)
+	u.SubmitLogs([]byte(payload))
+}
+
+func postLogs(logs string) {
+	u.SubmitLogs([]byte(logs))
+}
+
+func TestSuccessfulLogForwarding(t *testing.T) {
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		ExternalExtension()
+		wg.Done()
+	}()
+
+	// First invocation
+	requestId := uuid.New().String()
+	extensionReady()
+	extensionPlatformStart(requestId)
+	extensionInvoke(requestId)
+
+	messages := []string{"1 invocation\n", "2 invocation \n"}
+	postLogs(fmt.Sprintf(`[
+		{
+			"type": "function",
+			"record": "%s"
+		},
+		{
+			"type": "function",
+			"record": "%s"
+		}
+	]`, messages[0], messages[1]))
+
+	validationData := getValidations()
+
+	// Validate we received the request id we thought we would
+	if validationData.RequestId != requestId {
+		t.Errorf("Expected requestId %s Received %s", requestId, validationData.RequestId)
+	}
+
+	// Validate we received the logs we thought we would
+	for _, payload := range validationData.Logs {
+		if payload.Name != functionName {
+			t.Errorf("Expected function name %s Received %s", functionName, payload.Name)
+			for index, message := range payload.Messages {
+				if message.Message != messages[index] {
+					t.Errorf("Expected log message %s Received %s", message.Message, messages[index])
+				}
+			}
+		}
+	}
+
+	extensionPlatformRuntimeDone(requestId)
+	// Reset  validations
+	resetValidations()
+
+	// Second invocation
+	requestId2 := uuid.New().String()
+	extensionPlatformStart(requestId2)
+	extensionInvoke(requestId2)
+
+	messages2 := []string{"3 invocation\n", "4 invocation \n"}
+	postLogs(fmt.Sprintf(`[
+		{
+			"type": "function",
+			"record": "%s"
+		},
+		{
+			"type": "function",
+			"record": "%s"
+		}
+	]`, messages2[0], messages2[1]))
+
+	validationData2 := getValidations()
+
+	// Validate we received the request id we thought we would
+	if validationData2.RequestId != requestId2 {
+		t.Errorf("Expected requestId %s Received %s", requestId2, validationData2.RequestId)
+	}
+
+	// Validate we received the logs we thought we would
+	for _, payload := range validationData2.Logs {
+		if payload.Name != functionName {
+			t.Errorf("Expected function name %s Received %s", functionName, payload.Name)
+			for index, message := range payload.Messages {
+				if message.Message != messages[index] {
+					t.Errorf("Expected log message %s Received %s", message.Message, messages2[index])
+				}
+			}
+		}
+	}
+
+	// End execution
+	extensionPlatformRuntimeDone(requestId2)
+	extensionShutdown(requestId2)
+	wg.Wait()
+}
+
+func TestRuntimeDoneBeforeInvoke(t *testing.T) {
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		ExternalExtension()
+		wg.Done()
+	}()
+
+	// First invocation
+	requestId := uuid.New().String()
+	extensionReady()
+	extensionPlatformStart(requestId)
+	extensionInvoke(requestId)
+	extensionPlatformRuntimeDone(requestId)
```

Comment:
When doing this, we should confirm that it follows with `next` request being triggered

Reply:
This is already being testing by the second invocation right? If the next request was never triggered again the second invocation data would never come through correct?

Reply:
>  If the next request was never triggered again the second invocation data would never come through correct?

Yes, it's the reason, we should hold off with sending any further events until we record that `next` event was issued.

It's how it's done in JS tests: https://github.com/serverless/console/blob/e7e2cc667970bbc9a854fe47ef91157bfc6d1602/node/packages/aws-lambda-otel-extension/test/unit/external/index.test.js#L663

Reply:
Ahh okay got it 👍 I did this a little differently since that won't exactly translate to these tests but I am making sure we are receiving the expected number of calls to next.